src/ServiceProvider/BindingClosureFactory.php

```diff
@@ -69,6 +70,38 @@ public function createAutoWireProxyClosure($className, callable $parameterFactor
         };
     }
 
+    /**
+     * Create a proxy for an existing service definition. As opposed to autowired proxy definitions which are defined
+     * by the service itself, these proxies are defined by the client allowing clients to request proxied versions of
+     * otherwise non-lazy service definitions.
+     * This is preferable for services that should otherwise be initialised on construction (for example, those with
+     * complex dependency graphs that should normally be covered by service definition tests).
+     * @param Container $app
+     * @param string $serviceName The name of the service in the container.
+     * @param string $serviceClassName The FQCN of the service being proxied
+     * @return \ProxyManager\Proxy\VirtualProxyInterface
+     */
+    public function createServiceProxy(Container $app, string $serviceName, string $serviceClassName)
+    {
+        return $this->createProxy(
+            $serviceClassName,
+            function (Container $app) use ($serviceName, $serviceClassName) {
+                $service = $app->offsetGet($serviceName);
+                if (! ($service instanceof $serviceClassName)) {
+                    $invalidClassName = get_class($service);
+                    throw new \RuntimeException(
+                        "Invalid proxied/lazy service definition: tried to proxy '$serviceName' as an instance ".
+                        "of '$serviceClassName', but actually received an instance of '$invalidClassName' when retrieving ".
+                        "'$serviceName' from the container. To fix this, find the '\$this->getLazy($serviceName)' service ".
+                        "binding and make sure it specifies the actual class name that will be returned by that service."
+                    );
```

Comment:
I am wondering if in this situation we should use the [Error](https://www.php.net/manual/en/class.error.php) instead. Just being mindful of catch-all statements and possible recovery modes that clients might not be aware of, e.g. it is not a concern of the client as to how a particular class is wired together.

Reply:
Anyone using "catch all" statements is responsible for actually being able to catch all, so generally nobody should be catching all. 
Generic exceptions (i.e `\Exception` and `\RuntimeException` etc) should almost never be caught outside of tooling. 

As for `\Error` and it's non-exception variants, these are for internal PHP errors only and documented as such. We shouldn't be throwing these from our own code.